### PR TITLE
Breakdown the help command per feature

### LINKED ISSUE #180: Add kw help per feature
We had a lot of new features on kw, as a result, our `kw help` does not scale well for the amount of information that we need to display. I suggest that we copy the way that git handles help per feature (e.g., `git commit -h`), for example:
```bash
kw deploy -h
kw configm -h
```

### PR DESCRIPTION
**Pull request by alan-barzilay**
This PR breaks down the main help function, now it works in a similar manner to the help command in git.

I also updated the kw help to include all sub-commands that were missing.

Closes #180

### CODE REVIEW

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi @alan-barzilay ,

First of all, thanks a lot for your patch! Your patch is a huge step to make kw more user-friendly. I really appreciate your effort in this area.

I recommend you to see my review per commit instead of seeing all the comments together. However, I have some suggestions:
1. Make your series focused only on adding and improving the helper functions. Do not add sidequests to your series.
2. Keep each helper function per file. For example, instead of adding the statistics_help() in the help.sh, add it to statistics.sh and avoid one extra include. This makes much more sense because we want to keep each functionality isolated in the specific file. Let's keep the help.sh file only for the overall command summary and for other function helpers.
3. Take care in the way that you split your commits. Make commits per functionalities; in other words, your series can be summarized in two or three commits. Maybe one commit for prepare the ground and another commit that add the helper function per file.
4. I think your series added commits from other changes, make sure that you do not add unnecessary commits in your PR. I recommend you create a new branch based on the latest code from the unstable branch per feature.
5. Do not handle --help in the same way as -h. In your series, just use `-h`; I want to use the same approach that git has for the documentation. Usually, git use --help to open the man page for the specific feature; try git add -h and git add --help. P.s.: You don't need to handle --help, only -h.

I'm looking forward for your new series :)

File: src/help.sh [hunk 1]
```diff
@@ -10,21 +10,32 @@ function kworkflow-help()
        "\t Remote - machine reachable via the network"
 
   echo -e "\nCommands:\n" \
+
     "\tinit - Initialize kworkflow config file\n" \
     "\tbd - Build and install modules\n" \
+    "\tboot,bo - TODO\n" \
```
[the remark this answers is not in the document]
Comment (alan-barzilay) [reply src/help.sh:17]: I see, I didn't know. Maybe we should open an issue about removing it? I though that this option was still supported, I even opened an issue about adding it to the documentation  #214 haha If there is already an issue, I couldn't find it.

File: src/help.sh [hunk 2]
```diff
@@ -24,34 +24,50 @@ function kworkflow-help()
...
     "\tversion,--version,-v - show kw version\n" \
     "\tman - Show manual\n"
+}
 
+function build_help()
+{
   echo -e "kw build:\n" \
     "\tbuild - Build kernel \n" \
     "\tbuild [--menu|-n] - Open kernel menu config\n" \
+}
```
[the remark this answers is not in the document]
Comment (alan-barzilay) [reply src/help.sh]: done

File: kw [hunk 3]
```diff
@@ -199,7 +199,7 @@ function kw()
         . "$KW_LIB_DIR/help.sh" --source-only
 
         complain "Invalid option"
-        kworkflow-help
+        kworkflow_help
```
[the remark this answers is not in the document]
Comment (alan-barzilay) [reply kw]: done, see #226

File: src/help.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (rodrigosiqueira) [reply src/help.sh]: Maybe we can open an issue to delete this option.

**Comment by rodrigosiqueira** [comment]
Hi @alan-barzilay 
Your patch is almost perfect. I have few comments:
1. Take care with extra spaces at the end of lines; use `git log -p`, and you will see it. Also, do not add extra unnecessary lines in the code (see config_manager_help).
2. Your commit message does not describe well your work. 
3.

File: src/help.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (alan-barzilay) [reply src/help.sh]: done, #229

**Comment by alan-barzilay** [comment]
@rodrigosiqueira I have updated the series, please let me know if there is anything else to change

**Comment by rodrigosiqueira** [comment]
Hi @alan-barzilay 
I did a few changes in your commits and applied them to the unstable branch. See:

https://github.com/kworkflow/kworkflow/commit/cd9463d4e30c6f400821b5d3f9586cf667da120a

**Pull request closed without merging**